src/connect_to_peers.rs

```diff
@@ -108,6 +109,27 @@ async fn check_if_connection_is_allowed(
         return InternalConnectionStatus::Refused(ConnectionRefusedReason::BadStanding);
     }
 
+    if let Some(time) = global_state
+        .net
+        .last_disconnection_time_of_peer(*peer_address)
+    {
+        if SystemTime::now()
+            .duration_since(time)
+            .is_ok_and(|d| d < cli_arguments.reconnect_cooldown)
+        {
+            info!(
+                "Refusing incoming connection from {peer_address} \
+                 due to reconnect cooldown ({cooldown} seconds).",
+                cooldown = cli_arguments.reconnect_cooldown.as_secs(),
+            );
```

Comment:
The function `check_if_connection_is_allowed` that you're modifying here is invoked for both incoming and outgoing connections. But the log messages makes the assumption that you're handling an incoming connection.